commit/src/adapters/extension_mmcs.rs

```diff
@@ -49,7 +49,7 @@ where
             .into_iter()
             .map(|row| {
                 row.chunks(EF::DIMENSION)
-                    .map(EF::from_basis_coefficients_slice)
+                    .map(|chunk| EF::from_basis_coefficients_slice(chunk).unwrap()) // unwrap is safe as each chunk has length EF::DIMENSION
```

Comment:
Nit: `Option` implements `IntoIterator`, so we can just flatten:
```suggestion
                    .flat_map(EF::from_basis_coefficients_slice) // unwrap is safe as each chunk has length EF::DIMENSION
```

Reply:
Hmm, we immediately collect after this so I think it is probably fine?

E.g. Also the upper bound given by size_hint should be the correct value.